jormungandr/src/diagnostic.rs

```diff
@@ -0,0 +1,73 @@
+use crate::start_up::Error;
+use std::fmt::{self, Display, Formatter};
+
+#[cfg(unix)]
+pub type DiagnosticError = nix::Error;
+
+#[cfg(not(unix))]
+custom_error! {pub DiagnosticError {} = "diagnostic error"}
```

Comment:
It's OK to do it this way if `DiagnosticError` is meant for internal consumption, otherwise I think there might be an API mismatch problems on different platforms and so the error would better be a newtype.